src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -50,117 +59,144 @@ public String getUrlForPage(String wikiName, String pageName) {
+  }
+
   public String getUrlForPath(String wikiName, String wikiPath) {
     return addPathToUrl(getUrlForWiki(wikiName), wikiPath);
   }
 
-  public String getUrlForPath(String wikiName, String wikiPath, EnvType envType) {
-    return addPathToUrl(getUrlForWiki(wikiName, envType), wikiPath);
+  public String getUrlForPath(String wikiPath) {
+    return getUrlForPath(Configuration.getWikiName(), getLanguageForWiki(), wikiPath);
   }
 
   protected String addPathToUrl(String url, String path) {
-
-    String outputUrl = (!path.startsWith("/")) ? String.format("%s/%s", url, path): String.format("%s%s", url, path);
+    HttpUrl.Builder urlBuilder = HttpUrl.parse(url).newBuilder();
+    urlBuilder.addEncodedPathSegments(path.startsWith("/") ? path.replaceAll("^/", "") : path).build();
 
     String qs = Configuration.getQS();
     if (StringUtils.isNotBlank(qs)) {
-      outputUrl = appendQueryStringToURL(outputUrl, qs);
+      urlBuilder.encodedQuery(qs);
     }
-    return outputUrl;
+    return urlBuilder.build().toString();
   }
 
-  public String getUrlForPath(String wikiPath) {
-    return getUrlForPath(Configuration.getWikiName(), wikiPath);
+  public String getUrlForWiki() {
+    return getUrlForWiki(Configuration.getWikiName(), getLanguageForWiki(), false);
   }
 
-  public String getUrlForWiki() {
-    return getUrlForWiki(Configuration.getWikiName(), false);
+  public String getUrlForWiki(String wikiName, String language, boolean addWWW) {
+    return getUrlForWiki(wikiName, language, addWWW, envType);
   }
 
   public String getUrlForWiki(String wikiName) {
     return getUrlForWiki(wikiName, false);
   }
 
-  public String getUrlForWiki(String wikiName, EnvType envType) {
-    return getUrlForWiki(wikiName, false, envType);
-  }
-
-  public String getUrlForWiki(boolean addWWW) {
-    return getUrlForWiki(Configuration.getWikiName(), addWWW);
-  }
-
   public String getUrlForWiki(String wikiName, boolean addWWW) {
     return getUrlForWiki(wikiName, addWWW, envType);
   }
 
 
   public String getUrlForWiki(String wikiName, boolean addWWW, EnvType envType) {
+    return getUrlForWiki(wikiName, getLanguageForWiki(), addWWW, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language) {
+    return getUrlForWiki(wikiName, language, false, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language, boolean addWWW, EnvType envType) {
     final String wikiaName = getWikiaGlobalName(wikiName);
 
-    String www = "";
-    if (addWWW) {
-      www = "www.";
+    if (language == null || wikiName == null)
+      throw new NullPointerException("Wikia name and language are required");
+
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
+
+    String www = addWWW ? "www." : "";
+    String host = getFormattedWikiHost(www, wikiaName, envType, language);
+
+//    if (!("en").equals(language) && forceLanguageInPath) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)